src/modules/mavlink/mavlink_main.cpp

```diff
@@ -2568,50 +2467,21 @@ Mavlink::task_main(int argc, char *argv[])
 
 		_events.update(t);
 
-		/* pass messages from other UARTs */
+		/* pass messages from other instances */
 		if (get_forwarding_on()) {
 
-			bool is_part;
-			uint8_t *read_ptr;
-			uint8_t *write_ptr;
-
-			pthread_mutex_lock(&_message_buffer_mutex);
-			int available = message_buffer_get_ptr((void **)&read_ptr, &is_part);
-			pthread_mutex_unlock(&_message_buffer_mutex);
-
-			if (available > 0) {
-				// Reconstruct message from buffer
-
-				mavlink_message_t msg;
-				write_ptr = (uint8_t *)&msg;
-
-				// Pull a single message from the buffer
-				size_t read_count = available;
-
-				if (read_count > sizeof(mavlink_message_t)) {
-					read_count = sizeof(mavlink_message_t);
-				}
-
-				memcpy(write_ptr, read_ptr, read_count);
-
-				// We hold the mutex until after we complete the second part of the buffer. If we don't
-				// we may end up breaking the empty slot overflow detection semantics when we mark the
-				// possibly partial read below.
-				pthread_mutex_lock(&_message_buffer_mutex);
-
-				message_buffer_mark_read(read_count);
-
-				/* write second part of buffer if there is some */
-				if (is_part && read_count < sizeof(mavlink_message_t)) {
-					write_ptr += read_count;
-					available = message_buffer_get_ptr((void **)&read_ptr, &is_part);
-					read_count = sizeof(mavlink_message_t) - read_count;
-					memcpy(write_ptr, read_ptr, read_count);
-					message_buffer_mark_read(available);
-				}
-
-				pthread_mutex_unlock(&_message_buffer_mutex);
+			mavlink_message_t msg{};
```

Comment:
You don't strictly need to initialize this one
```suggestion
			mavlink_message_t msg;
```

Reply:
Correct! I'll change it.